### PR TITLE
Support built-in plugin operator with feature flag

### LINKED ISSUE #10939: UI changes to support built-in plugin operator
Related: SURE-6314

- Previously to manage extensions the ui-plugin-operator and ui-plugin-operator-crd charts needed to be installed
- Following https://github.com/rancher/rancher/pull/44230 these are now no longer required
- This affects how the UI sets up extension management

Initial appraisal ...  we need to....
1. update ui to handle initially installed and enabled operator, disabled via existing feature flag
    - At the moment the extension page checks for the operator service to determine if the charts should be installed... or can be uninstalled. The service no longer exists (shell/pages/c/_cluster/uiplugins/index.vue updateInstallStatus)
    - disabling the FF would be the same as uninstalling the charts. What does this mean for installed plugins?
1. update ui to notify and add extension catalogs (helm repos)
1. update tests given above
1. confirm that the existing method for air-gapped installs, used to bring in Rancher Extensions, still works

### PR DESCRIPTION
**Pull request by jordojordo**
### Summary
Fixes #10939 

This removes all mentions to enabling the ui-plugin-operator from the Extensions page and replaces the check for the operator service to a check for the `uiextensions` feature flag.

### Occurred changes and/or fixed issues

- RemoveUIPlugins.vue is no longer needed
- SetupUIPlugins.vue is now condensed now that the flow of enabling the feature flag is much different than installing the operator chart.
  - If the flag is not enabled the user will see a message explaining that Rancher will need to be restarted after enabling, and a button that leads the user to the feature flag settings page.
- The action for disabling extension support has been removed.

### Screenshot/Video

___Feature flag disabled___
extesion-flag-disabled

### Checklist
- [x] The PR is linked to an issue and the linked issue has a Milestone, or no issue is needed
- [x] The PR has a Milestone 
- [x] The PR template has been filled out
- [x] The PR has been self reviewed 
- [x] The PR has a reviewer assigned
- [x] The PR has automated tests or clear instructions for manual tests and the linked issue has appropriate QA labels, or tests are not needed
- [x] The PR has reviewed with UX and tested in light and dark mode, or there are no UX changes

### CODE REVIEW

**Review by aalves08 — COMMENTED** [review]
Gave this a test and looks good. Codewise, also lgtm. You do have the unit test and an e2e test failing in the pipelines... I just re-ran the e2e step again

**Comment by jordojordo** [comment]
I updated the unit tests to account for the feature flag, the Kubewarden e2e tests pass when running locally so I'll re-run them once more.

**Comment by aalves08** [comment]
@jordojordo @richard-cox ,for context, here's the capture of the error in our pipelines that meant the usage of `cy.once('uncaught:exception', () => false)` in `kubewarden.spec`:

Screenshot 2024-05-23 at 11 00 57

We cannot reproduce this locally (running e2e test on our machines) so we aren't exactly sure what is going on here. Weirdly the `extensions.spec` `before` hook is almost similar to the one in `kubewarden.spec` and there's no trouble there (never hit any error on our pipelines).

I know that this isn't ideal but after so many tries, we still haven't figured out where the exact problem lies 🙏

File: shell/pages/c/_cluster/uiplugins/index.vue [hunk 1]
```diff
@@ -148,6 +146,10 @@ export default {
     ...mapGetters({ uiErrors: 'uiplugins/errors' }),
     ...mapGetters({ theme: 'prefs/theme' }),
 
+    extensionsFeatureFlag() {
```
Comment (richard-cox) [inline shell/pages/c/_cluster/uiplugins/index.vue:149]: this isn't used

File: shell/pages/c/_cluster/uiplugins/index.vue [hunk 2]
```diff
@@ -488,46 +481,41 @@ export default {
...
+      let hasFeatureFlag;
 
       try {
-        const service = await this.$store.dispatch('management/find', {
-          type: SERVICE,
-          id:   `${ UI_PLUGIN_NAMESPACE }/ui-plugin-operator`,
-          opt:  { force: true },
+        const flag = await this.$store.dispatch('management/find', {
+          type: MANAGEMENT.FEATURE,
+          id:   'uiextension'
```
Comment (richard-cox) [inline shell/pages/c/_cluster/uiplugins/index.vue:497]: 'uiextension' can be added alongside the other feature flags in `shell/store/features.js`

there's a convenience way to get them as well `this.$store.getters['features/get'](STEVE_CACHE);`

File: shell/pages/c/_cluster/uiplugins/index.vue [hunk 3]
```diff
@@ -488,46 +481,41 @@ export default {
...
         });
 
-        hasService = !!service;
+        hasFeatureFlag = flag.spec?.value || flag.spec?.value !== false;
```
Comment (richard-cox) [inline shell/pages/c/_cluster/uiplugins/index.vue:500]: shouldn't this be `flag.spec?.value === true`?

File: shell/assets/translations/en-us.yaml [hunk 4]
```diff
@@ -4318,17 +4318,17 @@ plugins:
     title: Extension support is not enabled
     prompt:
       cant: Automatic installation is not available - required Helm Charts could not be found
-      can: You need to install the Extension Operator
+      can: You need to enable the Extensions feature flag in Global Settings. Once enabled, Rancher will require a restart.
```
Comment (richard-cox) [inline shell/assets/translations/en-us.yaml:4321]: this should receive the name of the FF and capitalise the resource, so

"You need to enable the `{ uiextension }` Feature Flag in Global Settings. Once ena...."

File: shell/assets/translations/en-us.yaml [hunk 5]
```diff
@@ -4318,17 +4318,17 @@ plugins:
...
       official: The Official Rancher Extensions Repository provides extensions published by Rancher
       partners: The Partners Extensions Repository provides extensions published by Rancher Partners
       community: The Community Extensions Repository provides extensions published by Rancher Community
     install:
       title: Enable Extension Support?
-      prompt: This will install the Helm charts needed to enable Extension support
+      prompt: This will lead you to the page where you can enable the feature flag needed to enable Extension support.
+      featuresButton: Go to Feature Flag settings
```
Comment (richard-cox) [inline shell/assets/translations/en-us.yaml:4328]: `Go to Feature Flags`

File: shell/assets/translations/en-us.yaml [hunk 6]
```diff
@@ -4318,17 +4318,17 @@ plugins:
...
       cant: Automatic installation is not available - required Helm Charts could not be found
-      can: You need to install the Extension Operator
+      can: You need to enable the Extensions feature flag in Global Settings. Once enabled, Rancher will require a restart.
       official: The Official Rancher Extensions Repository provides extensions published by Rancher
       partners: The Partners Extensions Repository provides extensions published by Rancher Partners
       community: The Community Extensions Repository provides extensions published by Rancher Community
     install:
       title: Enable Extension Support?
-      prompt: This will install the Helm charts needed to enable Extension support
+      prompt: This will lead you to the page where you can enable the feature flag needed to enable Extension support.
```
Comment (richard-cox) [inline shell/assets/translations/en-us.yaml:4327]: `Feature Flag`

File: shell/pages/c/_cluster/uiplugins/__tests__/SetupUIPlugins.test.ts [hunk 7]
```diff
@@ -1,128 +1,74 @@
...
-} from '@shell/config/uiplugins';
-import SetupUIPlugins from '@shell/pages/c/_cluster/uiplugins/SetupUIPlugins.vue';
-
-describe('component: SetupUIPlugins', () => {
-  const localVue = createLocalVue();
+import { mount } from '@vue/test-utils';
 
-  localVue.use(Vuex);
-
-  it('should NOT SHOW a checkbox to install official Rancher repo if NOT prime', async() => {
```
Comment (richard-cox) [inline shell/pages/c/_cluster/uiplugins/__tests__/SetupUIPlugins.test.ts:14]: Can this condition be checked on the other modal that adds helm repos?

File: cypress/e2e/tests/pages/extensions/kubewarden.spec.ts [hunk 8]
```diff
@@ -8,25 +8,30 @@ const extensionName = 'kubewarden';
 
 describe('Kubewarden Extension', { tags: ['@extensions', '@adminUser'] }, () => {
   before(() => {
+    // needed to swallow unreproducible error while running tests locally
+    // more info in https://github.com/rancher/dashboard/pull/11036
+    cy.once('uncaught:exception', () => false);
```
Comment (richard-cox) [inline cypress/e2e/tests/pages/extensions/kubewarden.spec.ts:13]: can we print out the e.message and use `catchTargetPageException` to catch the specific case that's the issue?

**Review by richard-cox — CHANGES_REQUESTED** [review]
- On a fresh rancher going to Extensions and  adding the partner repo via the banner and modal... none of the new extensions show up in the list. On refresh they (it) shows ok
- After disabling and enabling the feature flag the banner `There are new extensions repositories available. To enable these repositories, click the button on the right.` is shown, however clicking on button shows the modal with the partner charts already installed. Haven't traced through code, was this a bug from before as well?

File: shell/pages/c/_cluster/uiplugins/__tests__/SetupUIPlugins.test.ts (hunk 7 above)
Comment (jordojordo) [reply shell/pages/c/_cluster/uiplugins/__tests__/SetupUIPlugins.test.ts:14]: This condition is being checked in the [AddExtensionRepos.test.ts](https://github.com/rancher/dashboard/blob/c3cbd738b8b1a4142b7eab7442991f83ffb3f0ed/shell/pages/c/_cluster/uiplugins/__tests__/AddExtensionRepos.test.ts#L32)

File: cypress/e2e/po/pages/extensions/kubewarden.po.ts [hunk 9]
```diff
@@ -1,5 +1,5 @@
 import PagePo from '@/cypress/e2e/po/pages/page.po';
-import ExtensionsPagePo from '@/cypress/e2e/po/pages/extensions.po';
+// import ExtensionsPagePo from '@/cypress/e2e/po/pages/extensions.po';
```
Comment (richard-cox) [inline cypress/e2e/po/pages/extensions/kubewarden.po.ts:2]: comment only. i'm really eager to get this in, the extension based tests have been real flaky recently and i think this PR will help. in a future commit though can you remove these commented out lines, or add a comment why they should stay?

**Review by richard-cox — APPROVED** [review]

**Pull request merged**